### ISSUE TITLE
win10 64bits - spotty.exe crashes

### ISSUE THREAD

**Issue by chewiiie** [marcelveldt/plugin.audio.spotify#80]
Each time the plugin starts, I get the popup error message "spotty.exe stopped working" and it repeats indefinitly. The only way is to disable the plugin to be able to use kodi.
Windows and Kodi 17.6 are up-to-date; using your 1.0.1 mirror. I also tried to reinstall the plugin. No luck.
Is that a known issue ?

**Comment by sohilnandu** [comment]
This happened to me as well and had been bugging me for a long time. I finally dug into it and it seems like the plugin is trying to receive an authentication token before you have supplied Spotify credentials. This causes spotty.exe to keep crashing as there are no credentials. I tried to enter my credentials from the settings of the Spotify plugin but every time spotty.exe crashed it would reset my credentials to blank and I couldn't enter them fast enough before spotty.exe would crash again.

Hopefully this info is useful for someone that fully understands how the code works and can fix this issue.

I found a workaround just to make Kodi and the Spotify plugin usable again. You can hardcode your username and password in the plugin's util.py file. Then restart Kodi and it will be able to retrieve an authentication token. Now you can go to the plugin's settings and fill in your credentials where they belong. Don't forget to edit the util.py file again to remove your hardcoded credentials since they do not belong there at all. Restart Kodi again and all should be working fine.

**Comment by chewiiie** [comment]
Thanks a lot guys.
However i just updated to 1.1.56 version and spotty.exe keeps crashing
again.
Is there a log file somewhere I could send you to help ?

**Comment by sohilnandu** [comment]
Thanks for taking a look Marcel.
I think that it is a race condition when the user does not have any credentials stored in the plugin to begin with. Librespot is called without any credentials which causes the "spotty.exe has stopped working" message and the spotify plugin tries to call it again and again before the user has any time to provide their credentials. 
Is this plausible?

**Comment by chewiiie** [comment]
My Spotify credentials are already set in the plugin settings if that's
what you're thinking of.

**Comment by chewiiie** [comment]
Hello

It still does not work. I've again removed the extension but this time I
deleted also manually the plugins directory and the zip stored in cache.
Then I reinstalled and re-entered my credentials. But the spotty.exe
process still crashes continuously !!.
Any idea ?
Sohil, could you explain where in the utils.py you entered your login
informations ?

Thanks

2018-04-24 8:59 GMT+02:00 Chewbacca <chewbacca@lanproject.net>:

> My Spotify credentials are already set in the plugin settings if that's
> what you're thinking of.
>

**Comment by chewiiie** [comment]
Ok thanks for your quick answer. Unfortunately I had to uninstall the app.
It makes Kodi unusable.

**Comment by ryancpwalsh** [comment]
Note, I confirm the same behaviour on Win 7x64 with Kodi 17.6

The add-on within Kodi is working, music is playing, I can browse and search etc. - its fantastic.  But very frequently, spotty.exe crashes and the windows dialogue box notifying me of such minimizes Kodi.  Music continues to play, I can click ok on the notification box and go back into my minimized Kodi, no problem.

Generally when listening to music, I shut off my screen, so its not a huge deal breaker currently, but the crashing and notifications within windows keep from leaving Kodi on screen and enjoying visualizations etc.

Not sure what logs to provide, let me know, I'm happy to help participate in troubleshooting.

**Comment by eulores** [comment]
I had a similar issue of spotty.exe crashing. Checking the debug logs it seems as if the passed command line to spotty.exe is rather long! Might that be the issue for the crashes? Kodi was installed from the Windows store.
Once I reinstalled Kodi in a classical way (downloading and starting the installer) into a short directory, spotify plugin worked without any problems! No more crashes. Problem got fixed for me :-)

**Comment by chewiiie** [comment]
Wow thanks for the hint. I'm gonna give it a try !

**Comment by kimkoldtoft** [comment]
**SOLUTION!!!**

For me this was a really stupid small thing..

Seems that you need to login with your email address and not your user name.

Logging in with username works fine and I could see my music and everything, but spotty.exe kept crashing. But after using my email as username, it works fine.

Your welcome :)

Great addon Thanks.

**Comment by themalnz** [comment]
nope, not a solution (at all?) … still crashes regardless of login type used (username / handle or email address) … clearly the issue is rooted more than just "I used the wrong user credential" yes?

**Comment by chewiiie** [comment]
Reinstall kodi without using the windows store. When kodi is installed in a shorter path, like in program files, the plugin works.
